test/js-api/js-promise-integration/js-promise-integration.any.js

```diff
@@ -403,7 +401,7 @@ promise_test(async (t) => {
   let instance = builder.instantiate();
   let wrapper = WebAssembly.promising(instance.exports.test);
 
-  promise_rejects(t, RangeError, wrapper(),/Maximum call stack size exceeded/);
+  promise_rejects(t, new Error(), wrapper(), /Maximum call stack size exceeded/);
```

Comment:
1. These regular expressions should be strings. (Already fixed in WPT version)
2. Why not RangeError?

Reply:
1. We are ignoring the string/regex match -- they are different between engines
2. I cannot find similar test in WPT, in SpiderMonkey stack limit fails with InternalError("too much recursion").

Reply:
Also note from https://www.w3.org/TR/wasm-js-api-2/#stack-overflow

```
ECMAScript doesn’t specify any sort of behavior on stack overflow; implementations have been observed
to throw RangeError, InternalError or Error. Any is valid here.
```